### PR TITLE
Reacthttp

### PR DESCRIPTION
**Pull request by esokullu**

### CODE REVIEW

File: src/Pho/Server/Rest/Router.php [hunk 1]
```diff
@@ -11,88 +11,111 @@
...
+                
+            case Dispatcher::FOUND:
+                $handler = $routeInfo[1];
+                $vars = $routeInfo[2];
+                $this->kernel->logger()->info("Found: ".$handler . " --- ". print_r($vars, true));
+                // ... call $handler with $vars
+                // @todo This must be implemented
+                list($controllerName, $methodName) = explode('::', $routeInfo[1]);
+                if (null === $controller = $this->controllers[$controllerName] ?? null) {
+                    throw new \RuntimeException("Controller $controllerName not found");
```
Comment (esokullu) [inline src/Pho/Server/Rest/Router.php]: should not throw error, instead should $this->controllers["kernel"]->fail (status 500) and log.

File: src/Pho/Server/Rest/Router.php [hunk 2]
```diff
@@ -11,88 +11,111 @@
...
+                }
+                if (! method_exists($controller, $methodName)) {
+                    throw new \RuntimeException("Method $methodName does not exist in controller $controllerName");
+                }
+                $response = new Response(200, AbstractController::HEADERS, null);
+                $response = call_user_func_array([ $controller, $methodName ], array_merge([ $request, $response ], $vars));
```
Comment (esokullu) [inline src/Pho/Server/Rest/Router.php:115]: I don't see how you deal with json vs jsonp case here. See AbstractController.php.

Comment (esokullu) [reply src/Pho/Server/Rest/Router.php:115]: cc/ @codekaar

File: src/Pho/Server/Rest/Router.php (hunk 1 above)
Comment (esokullu) [reply src/Pho/Server/Rest/Router.php]: cc/ @codekaar

Comment (sudo-barun) [reply src/Pho/Server/Rest/Router.php]: fixed
see 68a5d8d871c04b126b940ff691dcd9eeeaea0e90

File: src/Pho/Server/Rest/Router.php (hunk 2 above)
Comment (sudo-barun) [reply src/Pho/Server/Rest/Router.php:115]: json vs jsonp case is not handled here. Instead, it is handled in `AbstractController::respond`

**Pull request merged**